prober/icmp.go

```diff
@@ -122,9 +127,22 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 		return
 	}
 	level.Info(logger).Log("msg", "Writing out packet")
-	if _, err = socket.WriteTo(wb, ip); err != nil {
-		level.Warn(logger).Log("msg", "Error writing to socket", "err", err)
-		return
+	if socket6 != nil && len(module.ICMP.SourceIPAddress) > 0 {
+		// Set source address for IPv6.
```

Comment:
Why are you handling this in a different place than for v4?

Reply:
Checking for len() > 0 and parse errors is now in common code. 
The actual control message is v6 specific.